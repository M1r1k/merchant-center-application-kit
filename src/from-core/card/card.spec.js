import React from 'react';
import { shallow } from 'enzyme';
import Card from './card';

describe('rendering', () => {
  describe('with `raised` `type`', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<Card>{'text'}</Card>);
    });

    it('should output correct tree', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should apply the `type-raised` class name', () => {
      expect(wrapper).toHaveClassName('type-raised');
    });
  });

  describe('with `flat` `type`', () => {
    const props = { type: 'flat' };
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<Card {...props}>{'text'}</Card>);
    });

    it('should output correct tree', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should apply the `type-flat` class name', () => {
      expect(wrapper).toHaveClassName('type-flat');
    });
  });
});
