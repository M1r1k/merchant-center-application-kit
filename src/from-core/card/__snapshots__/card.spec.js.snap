// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering with \`flat\` \`type\` should output correct tree 1`] = `
<div
  className="container type-flat"
>
  text
</div>
`;

exports[`rendering with \`raised\` \`type\` should output correct tree 1`] = `
<div
  className="container type-raised"
>
  text
</div>
`;
